### DISCUSSION TITLE
How to override fileMatch of a regexManager preset?

### DISCUSSION THREAD

**Discussion by childnode** [renovatebot/renovate#21228] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.24.6

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

While it is easy to override the fileMatch [on named managers](https://github.com/renovatebot/renovate/discussions/10845), it's unevenly more difficult / impossible to do for a [regexManager (preset)](https://docs.renovatebot.com/presets-regexManagers/), is it?!

Example:
```
module.exports = {
    "extends": [
        "local>renovate/renovate-presets:regexPresetForSomeCase"
    ],
}
```

where preset `regexPresetForSomeCase` is defined as 
```
{
  "regexManagers": [
    {
      "fileMatch": [
        "someFile$"
      ],
      "matchStrings": [
        "find.*things"
      ]
    }
  ]
}
``` 

as possible for others things set by preset, I expect to get s.th. like ...
```
module.exports = {
    "extends": [
        "local>renovate/renovate-presets:regexPresetForSomeCase"
    ],
    "local>renovate/renovate-presets:regexPresetForSomeCase:regexManagers:0": {
       "fileMatch: [
          "someFile$",
          "someOtherFile$",
        ]
    }
}
```

even sufficient to me would be to re-use fileMatch in a regexManager from another manager like:
```
{
  "regexManagers": [
    {
      "extends": [ "gitlabci-include" ],
      "matchStrings": [
        "find.*things"
      ]
    }
  ]
}
```

### What you tried so far.

digging the documentation, e.g. https://docs.renovatebot.com/config-presets/#extending-from-a-preset , https://docs.renovatebot.com/key-concepts/presets/, https://docs.renovatebot.com/modules/manager/regex/ , https://docs.renovatebot.com/presets-regexManagers/, …

I can't read any named constraint but I can not read any other helpful ;(

digging code where `config.js` is parsed as a "function" https://github.com/renovatebot/renovate/blob/2b43f273bf0737ccf29b27cf55ecec69403bfdf1/lib/workers/global/config/parse/file.ts#L23 mind-gobbling if I can hack in some dynamic code to change it ... But I doubt I can't at this point at least because the extends / presets are not resolved yet

**Comment by rarkins** [comment, marked as answer]
Unfortunately I don't think it's possible or easily implementable

**Reply by childnode** [reply]
hi, thank you for that quick "indication" ;)
sad to hear that.

I extended my question by - if it would be possible to do it "reverse" in my case like?!
```
{
  "regexManagers": [
    {
      "extends": [ "gitlabci-include" ],
      "matchStrings": [
        "find.*things"
      ]
    }
  ]
}
```
where extends means not include a preset - see https://github.com/renovatebot/renovate/issues/12205 - but extend / include the `gitlabci-include` manager fileMatch rule

**Reply by rarkins** [reply]
Not possible right now, but might be possible to implement so that it's supported

**Reply by childnode** [reply]
can you give me an indication at which point? perhaps I can contribute a PR

**Reply by rarkins** [reply]
Normally, you aren't meant to use `extends` that way, and it may not be easy to do. But we make one exception that I can recall, and that is to enable our "monorepos" grouping.

An example monorepo is here: https://docs.renovatebot.com/presets-monorepo/#monorepoaspnet-aspnetwebstack
An corresponding group for it is here: https://docs.renovatebot.com/presets-group/#groupaspnet-aspnetwebstackmonorepo

The more I think about it, the less likely I think it is to be easy.

Can you describe _why_ you need this? e.g. are these both your own presets, or are you wanting to extend someone else's preset without copy pasting it? Is this a common need you have? And if not then why not copy paste it?
